---
title: "Zebrafish summary"
author: "Karissa"
date: "06/06/2020"
output: 
  html_document:
    toc: true
    toc_float: true
    code_folding: hide
editor_options: 
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# setup

```{r}
library(UpSetR)
library(RColorBrewer)
library(tidyverse)
library(magrittr)
library(pheatmap)
library(readxl)
library(biomaRt)
library(edgeR)
library(factoextra) 
```

# annotations
```{r}
ah <- AnnotationHub() %>%
  subset(species == "Danio rerio") %>%
  subset(dataprovider == "Ensembl") %>%
  subset(rdataclass == "EnsDb")
ensDb <- ah[["AH69169"]]
genes <- genes(ensDb) %>% 
  as_tibble
```

# gene sets
```{r}
KEGG <- 
  msigdbr("Danio rerio", category = "C2", subcategory = "CP:KEGG") %>% 
  left_join(genes %>% unchop(entrezid), by = c("entrez_gene" = "entrezid")) %>% 
  dplyr::filter(!is.na(gene_id)) %>%
  distinct(gs_name, gene_id, .keep_all = TRUE) %>% 
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")
```


# Read in the HMP values from each zebrafish study
```{r}

HMP_q96_6m <- readRDS("data/zebrafish/R_objects/psen1Q96_HMP.rds") %>% 
  mutate(mutation = "Q96_K97del/+", 
         gene = "psen1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)


sorl1_2way_HMP <-readRDS("data/zebrafish/R_objects/sorl1_2way_HMP.rds") %>% 
  mutate(mutation = "W1818*/+", 
         gene = "sorl1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)

sorl14way_HMP <- readRDS("data/zebrafish/R_objects/sorl14way_HMP.rds") %>% 
  mutate(mutation = coef, 
         gene = "sorl1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)


psen2_HMP <- readRDS("data/zebrafish/R_objects/psen2_HMP.rds")%>% 
  mutate(mutation = case_when(coef == "FAD" ~ "T141_L142delinsMISLISV/+", 
                              coef == "FS" ~ "N140fs/+"
                            ), 
         gene = "psen2") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)

psen1_T428del_HMP <- readRDS("data/zebrafish/R_objects/HMP_raw_KEGG_IRE.rds") %>% 
  mutate(mutation = case_when(coef == "fAI-like/+" ~ "fAI-like/+", 
                              coef == "EOfAD-like/+" ~ "T428del/+"
  ), 
  gene = "psen1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)

psen1_K97fs_HMP <- readRDS("data/zebrafish/R_objects/k97_HMP.rds") %>% 
  mutate(mutation = "K97fs/+", 
         gene = "psen1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)

Q96_hyp_HMP <- readRDS("~/Documents/AD-signature/data/zebrafish/R_objects/Q96_hyp_HMP.rds") %>% 
   mutate(mutation = "Q96_K97del/+ (hypoxia)", 
         gene = "psen1") %>% 
  dplyr::select(pathway, harmonic_p, harmonic_p_FDR, mutation, gene)

HMP_all <- 
  HMP_q96_6m %>% 
  bind_rows(sorl1_2way_HMP) %>% 
  bind_rows(sorl14way_HMP) %>% 
  bind_rows(psen1_T428del_HMP) %>% 
  bind_rows(psen2_HMP) %>% 
  bind_rows(psen1_K97fs_HMP) %>%
  bind_rows(Q96_hyp_HMP) %>% 
  mutate(`EOfAD-like` = case_when(
    mutation %in% c("N140fs/+", "fAI-like/+", "K97fs/+") ~ "No",
    TRUE ~ "Yes"
  )
  ) 

sigpaths <- HMP_all %>% 
  dplyr::filter(harmonic_p_FDR < 0.05) %>% 
  .$pathway %>% 
  unique

# plot heatmap
anno_mut <- 
  HMP_all %>% 
  dplyr::select(mutation, gene, `EOfAD-like`) %>% 
  unique %>% 
  mutate(Experiment = case_when(
    mutation %in% c("R122Pfs/+", "trans", "V1482Afs/+") ~ "Experiment 1", 
    mutation %in% c("T428del/+", "fAI-like/+") ~ "Experiment 2",
    grepl(mutation, pattern = "Q96_K97del/+") ~ "Experiment 3",
    mutation %in% c("W1818*/+") ~ "Experiment 4",
    mutation %in% c("N140fs/+", "T141_L142delinsMISLISV/+") ~ "Experiment 5",
    mutation %in% "K97fs/+" ~ "Experiment 6"
  )) %>% 
  column_to_rownames("mutation") 

sig_in_morethan1 <- 
  HMP_all %>% 
  dplyr::filter(grepl(pathway, pattern = "^KEGG_|^ire")) %>% 
  dplyr::filter(pathway %in% sigpaths) %>% 
  group_by(pathway) %>% 
  count(harmonic_p_FDR < 0.05 ) %>% 
  dplyr::filter(`harmonic_p_FDR < 0.05` == "TRUE" & n >= 2) %>% 
  .$pathway

anno_hmp <- 
  HMP_all %>% 
  dplyr::filter(pathway %in% sigpaths) %>% 
  dplyr::filter(pathway %in% sig_in_morethan1) %>% 
  dplyr::select(pathway, harmonic_p_FDR, mutation) %>% 
  mutate(harmonic_p_FDR = signif(harmonic_p_FDR, 2)) %>% 
  dplyr::filter(grepl(pathway, pattern = "^KEGG_|^ire")) %>% 
  spread(key = "mutation", value = "harmonic_p_FDR") %>% 
  column_to_rownames("pathway")

# png("output/HMP.png")
HMP_all %>% 
  dplyr::select(pathway, harmonic_p_FDR, mutation) %>% 
  dplyr::filter(grepl(pathway, pattern = "^KEGG_|^ire")) %>% 
  dplyr::filter(pathway %in% sig_in_morethan1) %>% 
  spread(key = "mutation", value = "harmonic_p_FDR") %>% 
  column_to_rownames("pathway") %>% 
  pheatmap(color = viridis::viridis_pal(option = "viridis", begin = 1, end = 0)(200), 
           # cluster_cols = TRUE,
           gaps_row = 3,
           display_numbers = anno_hmp, 
           clustering_method = "complete",
           border_color = "black",
           number_format = "%.2f" ,
           # number_color = "red",
           annotation_col = anno_mut,
           annotation_colors = list(
             `EOfAD-like` = c("Yes" = "firebrick3",
                              "Complex" = "grey25",
                              "No" = "turquoise4"),
             gene = c("psen1" = "seagreen2",
                      "psen2" = "orange2",
                      "sorl1" = "plum4")
           )
  )
#dev.off()

# write out as csv for S2 file
# HMP_all %>% write_csv2("output/harmonicmeanpvals.csv")

```

# Heatmap of oxphos

```{r}
logFCs <-
read_csv("data/zebrafish/R_objects/Normoxia_Q96mutant_vs_wt_limma (1).csv") %>% 
  mutate(mutation = "psen1 Q96_K97del/+", 
         gene = "psen1") %>% 
  dplyr::select(gene_id = ensembl, logFC, FDR, mutation, gene) %>% 
  bind_rows(
    topTable(readRDS("data/zebrafish/R_objects/Q96fit.rds"), 
             n = Inf, 
             coef = "hyp_6m_mutvWT") %>% 
      dplyr::select(gene_id = ensembl_gene_id , logFC, FDR = adj.P.Val) %>% 
      mutate(mutation = "psen1 Q96_K97del/+ (hypoxia)", 
             gene = "psen1")
  ) %>% 
  bind_rows(
    readRDS("data/zebrafish/R_objects/sorl1_4way_toptable.rds") %>% 
      bind_rows() %>% 
      dplyr::select(gene_id, logFC, FDR, mutation = coef) %>% 
      mutate( mutation = paste("sorl1", mutation),
        gene = "sorl1"
             )
  ) %>% 
  bind_rows(
    readRDS("data/zebrafish/R_objects/sorl1_2way_toptable.rds") %>% 
      bind_rows() %>% 
      dplyr::select(gene_id, logFC, FDR) %>% 
      mutate(mutation = "sorl1 W1818*/+", 
             gene = "sorl1")
  ) %>% 
  bind_rows(
    readRDS("data/zebrafish/R_objects/psen2_toptableRUV.rds") %>% 
      bind_rows() %>% 
      left_join(genes %>% as_tibble()) %>% 
      dplyr::select(gene_id, logFC, FDR, mutation = "coef") %>% 
      mutate(gene = "psen2", 
             mutation = case_when(
               mutation == "FS" ~ "psen2 N140fs/+", 
               mutation == "FAD" ~ "psen2 T141_L142delinsMISLISV/+"
             ))
  ) 
  bind_rows(
    readRDS("data/zebrafish/R_objects/psen1_3way_toptable.rds") %>% 
      bind_rows() %>% 
      dplyr::select(gene_id, logFC, FDR, mutation = "coef") %>% 
      mutate(gene = "psen1", 
             mutation = case_when(
               mutation == "EOfAD-like/+" ~ "psen1 T428del/+", 
               mutation == "fAI-like/+" ~ "psen1 W233fs/+"
             ))
  )
  bind_rows(K97.de.res %>% 
              mutate(mutation = "psen1 K97fs/+", 
                     gene = "psen1") %>% 
              dplyr::select(gene_id, logFC, FDR, mutation, gene)
            ) 

oxphos_anno <- KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION %>% 
  as_tibble() %>% 
  set_colnames("gene_id") %>% 
  left_join(genes) %>% 
  dplyr::select(gene_name, description) %>% 
  dplyr::distinct(gene_name, .keep_all = TRUE) %>% 
  mutate(complex = case_when(
    grepl(description, pattern = "NADH") ~ "Complex I", 
    grepl(description, pattern = "NDUFA4") ~ "Complex I", 
    grepl(description, pattern = "succinate dehydrogenase complex, subunit") ~ "Complex II",
    grepl(description, pattern = "ubiquinol-cytochrome c") ~ "Complex III", 
    grepl(description, pattern = "cytochrome c-1") ~ "Complex III", 
    grepl(description, pattern = "cytochrome b, mitochondrial") ~ "Complex III", 
    grepl(description, pattern = "cytochrome c oxidase subunit") ~ "Complex IV", 
    gene_name %in% c("cox10", "cox11", "cox15", "cox17") ~ "Complex IV", 
    grepl(gene_name, pattern = "mt-co") ~ "Complex IV", 
    grepl(gene_name, pattern = "COX5B") ~ "Complex IV", 
    grepl(description, pattern = "ATP") ~ "Complex V", 
  )
  ) %>% 
  dplyr::select(-description) %>% 
  column_to_rownames("gene_name")

oxphos_anno_list <- oxphos_anno %>% 
  rownames_to_column("gene_name") %>% 
  split(f = .$complex) %>% 
  lapply(magrittr::extract2, "gene_name")

oxphos_anno_colours <- brewer.pal(name = "Set1", n = 5)
names(oxphos_anno_colours) <- names(oxphos_anno_list) 
oxphos_anno_colours <- list(complex = oxphos_anno_colours)

# png("output/oxphos_.png", width = 80, height = 20, units = "cm", res = 800)  
# logFCs %>%   
#   dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>% 
#   left_join(genes %>% as_tibble()) %>% 
#   dplyr::distinct(gene_name, mutation, .keep_all = TRUE) %>% 
#   dplyr::select(gene_name, logFC, mutation) %>% 
#   spread(key = "mutation", value = "logFC") %>% 
#   na.omit %>% 
#   column_to_rownames("gene_name") %>% 
#   .[c(oxphos_anno_list$`Complex I`, 
#       oxphos_anno_list$`Complex II`,
#       oxphos_anno_list$`Complex III`,
#       oxphos_anno_list$`Complex IV`,
#       oxphos_anno_list$`Complex V`
#       ), ] %>% 
#   na.omit() %>% #.[c(oxphos_anno_list$`Complex I`, oxphos_anno_list$`Complex II`, oxphos_anno_list$`Complex III`),] %>% dim
#   t() %>% 
#   pheatmap(color = colorRampPalette(rev(brewer.pal(n = 11, 
#                                                    name = "RdBu")))(100), 
#            cellwidth = 10,
#            cellheight = 40,
#            cluster_cols = FALSE,
#            show_colnames = FALSE,
#            fontsize_row = 25,
#            angle_col = 315, 
#            annotation_col = oxphos_anno,
#            annotation_colors = oxphos_anno_colours,
#            gaps_col = c(40, 44, 52, 71),
#            breaks =  seq(-0.5, 0.5, length.out = 100),
#            main = "KEGG_OXIDATIVE_PHOSPHORYLATION")
#  dev.off()  
# png("output/oxphos_wcolnames.png", width = 95, height = 20, units = "cm", res = 400) 
 logFCs %>%   
  dplyr::filter(gene_id %in% KEGG$KEGG_OXIDATIVE_PHOSPHORYLATION) %>% 
  left_join(genes %>% as_tibble()) %>% 
  dplyr::distinct(gene_name, mutation, .keep_all = TRUE) %>% 
  dplyr::select(gene_name, logFC, mutation) %>% 
  spread(key = "mutation", value = "logFC") %>% 
  na.omit %>% 
  column_to_rownames("gene_name") %>% 
  .[c(oxphos_anno_list$`Complex I`, 
      oxphos_anno_list$`Complex II`,
      oxphos_anno_list$`Complex III`,
      oxphos_anno_list$`Complex IV`,
      oxphos_anno_list$`Complex V`
      ), ] %>% 
  na.omit() %>% #.[c(oxphos_anno_list$`Complex I`, oxphos_anno_list$`Complex II`, oxphos_anno_list$`Complex III`),] %>% dim
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 11, 
                                                   name = "RdBu")))(100), 
           cellwidth = 18,
           cellheight = 40,
           cluster_cols = FALSE,
           show_colnames = T,
           fontsize_col = 15,
           fontsize_row = 25,
           angle_col = 315, 
           annotation_col = oxphos_anno,
           annotation_colors = oxphos_anno_colours,
           gaps_col = c(40, 44, 52, 71),
           breaks =  seq(-0.5, 0.5, length.out = 100),
           main = "KEGG_OXIDATIVE_PHOSPHORYLATION")
 # dev.off()
```

# Heatmap of ribosome gene expression

```{r}
ribo_anno_list <- 
  KEGG$KEGG_RIBOSOME %>% 
  as_tibble() %>% 
  set_colnames("gene_id") %>% 
  left_join(genes) %>% 
  dplyr::select(gene_name, description) %>% 
  dplyr::distinct(gene_name, .keep_all = TRUE) %>% 
  mutate(subunit = case_when(
    grepl(description, pattern = "ribosomal protein L") ~ "Large subunit",
    grepl(description, pattern = "large") ~ "Large subunit",
    grepl(gene_name, pattern = "rsl24d1") ~ "Large subunit",
    grepl(gene_name, pattern = "uba52") ~ "Large subunit",
    grepl(description, pattern = "ribosomal protein S") ~ "Small subunit",
    grepl(gene_name, pattern = "fau") ~ "Small subunit",
    grepl(gene_name, pattern = "RPS17") ~ "Small subunit"
  )) %>% 
  dplyr::select(-description) %>% 
  split(f = .$subunit) %>% 
  lapply(magrittr::extract2, "gene_name")

ribo_anno <- 
  KEGG$KEGG_RIBOSOME %>% 
  as_tibble() %>% 
  set_colnames("gene_id") %>% 
  left_join(genes) %>% 
  dplyr::select(gene_name, description) %>% 
  dplyr::distinct(gene_name, .keep_all = TRUE) %>% 
  mutate(subunit = case_when(
    grepl(description, pattern = "ribosomal protein L") ~ "Large subunit",
    grepl(description, pattern = "large") ~ "Large subunit",
    grepl(gene_name, pattern = "rsl24d1") ~ "Large subunit",
    grepl(gene_name, pattern = "uba52") ~ "Large subunit",
    grepl(description, pattern = "ribosomal protein S") ~ "Small subunit",
    grepl(gene_name, pattern = "fau") ~ "Small subunit",
    grepl(gene_name, pattern = "RPS17") ~ "Small subunit"
  )) %>% 
  dplyr::select(-description) %>% 
  column_to_rownames("gene_name")

png("output/ribo_wcolnames.png", width = 78, height = 20, units = "cm", res = 400)  
logFCs %>%   
  dplyr::filter(gene_id %in% KEGG$KEGG_RIBOSOME) %>% 
  left_join(genes %>% as_tibble()) %>% 
  dplyr::distinct(gene_name, mutation, .keep_all = TRUE) %>% 
  dplyr::select(gene_name, logFC, mutation) %>% 
  spread(key = "mutation", value = "logFC") %>% 
  column_to_rownames("gene_name") %>% 
  .[c(ribo_anno_list$`Large subunit`, ribo_anno_list$`Small subunit`),] %>% 
  na.omit %>% 
  t() %>% 
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 11, 
                                                   name = "RdBu")))(100), 
           cellwidth = 20,
           cellheight = 40,
           fontsize_col = 18,
           cluster_cols = FALSE,
           show_colnames = T,
           fontsize_row = 25,
           angle_col = 315, 
           annotation_col = ribo_anno,
           gaps_col = 47,
           breaks =  seq(-0.5, 0.5, length.out = 100),
           main = "KEGG_RIBOSOME")
dev.off()
```

# cell type proportions

```{r}
sorl1_dge_RUV <- readRDS("data/zebrafish/R_objects/sorl1_dge_RUV.rds")

#biomart to get the conversion between mouse and zebrafish IDs
mart <- useMart("ensembl", "drerio_gene_ensembl")
getFromBiomart <- c("ensembl_gene_id", "mmusculus_homolog_associated_gene_name")
zf_id_2_mm_genename <- getBM(attributes = getFromBiomart,
                             mart = mart) %>% 
  as_tibble() %>% 
  set_colnames(c("gene_id", "mouse_gene_name"))

cell_type_markers <- 
  read_xls("data/cell_type_genes41586_2007_BFnature05453_MOESM11_ESM.xls") %>%
  as_tibble() %>%
  dplyr::select(c("Neuron-enriched", "Astrocyte-enriched", "Oligodendrocyte-enriched")) %>%
  gather(key = "cell_type", value = "mouse_gene_name") %>% 
  left_join(zf_id_2_mm_genename) %>% 
  na.omit %>% 
  split(f = .$cell_type) %>% 
  lapply(extract2, "gene_id")

cell_type_markers$`Microglia-enriched` <- 
  read_excel("data/microglia_oosterofetal.xlsx", sheet = 1) %>%
  rbind(read_excel("data/microglia_oosterofetal.xlsx", sheet = 2)) %>%
  rbind(read_excel("data/microglia_oosterofetal.xlsx", sheet = 3)) %>%
  .$Zebrafish.Ensembl.Gene.ID %>% 
  unique

# work out optimum number of clusters for kmeans 
# sorl1_dge_RUV %>% 
#   cpm(log=TRUE) %>%
#   as.data.frame() %>%
#   .[cell_type_markers$`Microglia-enriched`,] %>%
#   na.omit %>% 
#   fviz_nbclust(kmeans, method = "wss") 

annoForCelltypesSORL1 <- sorl1_dge_RUV$samples %>% 
  dplyr::select(new_sample_id, Genotype) %>% 
  column_to_rownames("new_sample_id")

png("output/zebrafish cell type proportions/microglia_sorl1_4way.png", width = 25, height = 10, units = "cm", res = 800)  
sorl1_dge_RUV %>%
  cpm(log=TRUE) %>%
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", sorl1_dge_RUV$samples$sample) %>% 
  left_join(sorl1_dge_RUV$samples) %>% 
  as_tibble() %>% 
  dplyr::select(gene_id, new_sample_id, logCPM) %>% 
  spread(key = "new_sample_id", value = "logCPM") %>% 
  column_to_rownames("gene_id") %>% 
  .[cell_type_markers$`Microglia-enriched`,] %>% 
  na.omit %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           annotation_col = annoForCelltypesSORL1,
           annotation_colors = list(Genotype = c(WT = "#ff6666", 
                                                 trans = "#CCFF66",
                                                 `R122Pfs/+` = "#5D2E8C", 
                                                 `V1482Afs/+` = "#2EC4B6" )),
           kmeans_k = 5,
           # cluster_rows = F, cluster_cols = F,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Microglia" 
  ) 
dev.off()

png("output/zebrafish cell type proportions/neurons_sorl14way.png", width = 25, height = 10, units = "cm", res = 800)  
sorl1_dge_RUV %>% 
  cpm(log=TRUE) %>%
  as.data.frame() %>% 
    rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", sorl1_dge_RUV$samples$sample) %>% 
  left_join(sorl1_dge_RUV$samples) %>% 
  as_tibble() %>% 
  dplyr::select(gene_id, new_sample_id, logCPM) %>% 
  spread(key = "new_sample_id", value = "logCPM") %>% 
  column_to_rownames("gene_id") %>% 
  .[cell_type_markers$`Neuron-enriched`,] %>% 
  na.omit %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           annotation_col = annoForCelltypesSORL1,
           annotation_colors = list(Genotype = c(WT = "#ff6666", 
                                                 trans = "#CCFF66",
                                                 `R122Pfs/+` = "#5D2E8C", 
                                                 `V1482Afs/+` = "#2EC4B6" )),
           kmeans_k = 5,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Neurons" 
  ) 
dev.off()

png("output/zebrafish cell type proportions/Astrocyte_sorl14way.png", width = 25, height = 10, units = "cm", res = 800)  
sorl1_dge_RUV %>% 
  cpm(log=TRUE) %>%
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", sorl1_dge_RUV$samples$sample) %>% 
  left_join(sorl1_dge_RUV$samples) %>% 
  as_tibble() %>% 
  dplyr::select(gene_id, new_sample_id, logCPM) %>% 
  spread(key = "new_sample_id", value = "logCPM") %>% 
  column_to_rownames("gene_id") %>% 
  .[cell_type_markers$`Astrocyte-enriched`,] %>% 
  na.omit %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           annotation_col = annoForCelltypesSORL1,
           annotation_colors = list(Genotype = c(WT = "#ff6666", 
                                                 trans = "#CCFF66",
                                                 `R122Pfs/+` = "#5D2E8C", 
                                                 `V1482Afs/+` = "#2EC4B6" )),
           kmeans_k = 5,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Astrocytes" 
  ) 
dev.off()

png("output/zebrafish cell type proportions/Oligodendrocytes_sorl14way.png", width = 25, height = 10, units = "cm", res = 800)  
sorl1_dge_RUV %>% 
  cpm(log=TRUE) %>%
  as.data.frame() %>% 
    rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", sorl1_dge_RUV$samples$sample) %>% 
  left_join(sorl1_dge_RUV$samples) %>% 
  as_tibble() %>% 
  dplyr::select(gene_id, new_sample_id, logCPM) %>% 
  spread(key = "new_sample_id", value = "logCPM") %>% 
  column_to_rownames("gene_id") %>% 
  .[cell_type_markers$`Oligodendrocyte-enriched`,] %>% 
  na.omit %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           annotation_col = annoForCelltypesSORL1,
           annotation_colors = list(Genotype = c(WT = "#ff6666", 
                                                 trans = "#CCFF66",
                                                 `R122Pfs/+` = "#5D2E8C", 
                                                 `V1482Afs/+` = "#2EC4B6" )),
           kmeans_k = 5,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Oligodendrocytes" 
  ) 
dev.off()

```

## Q96K97del cell types

Look at 6m and 24m

```{r}
# read in gene expression values for Q96_K97del
Q96_dge <- readRDS("data/zebrafish/R_objects/dge_all.rds")

Q96_anno <- Q96_dge$samples %>% 
  as_tibble() %>% 
  dplyr::select(sample, Age, Genotype, Hypoxia, Sex = Gender) %>% 
  mutate(Hypoxia = case_when(Hypoxia == "0" ~ "Normoxia", 
                             Hypoxia == "1" ~ "Hypoxia"), 
         Genotype = case_when(Genotype == "wt" ~ "WT", 
                              Genotype == "q96" ~ "Q96_K97del/+"), 
         Age = paste0(Age, "m")) %>% 
  column_to_rownames("sample")
Q96_annoColours  <- list(Genotype = c(WT = "#FF6666", 
                                      `Q96_K97del/+` = "#CCFF66"), 
                         Age = c(`6m` = "#88498F", 
                                 `24m` = "#779FA1"), 
                         Sex = c(male = "#6A7FDB", 
                                 female = "#57E2E5"), 
                         Hypoxia = c(Normoxia = "#ACBEA3", 
                                     Hypoxia = "#40476D")
                         )

png("output/zebrafish cell type proportions/q96 Oligodendrocytes.png", width = 25, height = 10, units = "cm", res = 800) 
Q96_dge %>% 
  cpm(log=TRUE) %>% 
  as.data.frame() %>% 
  .[cell_type_markers$`Oligodendrocyte-enriched`,] %>% 
  na.omit() %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           kmeans_k = 5,
           annotation_col = Q96_anno,
           annotation_colors = Q96_annoColours,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Oligodendrocytes" 
  ) 
dev.off()

png("output/zebrafish cell type proportions/q96 Neurons.png", width = 25, height = 10, units = "cm", res = 800) 
Q96_dge %>% 
  cpm(log=TRUE) %>% 
  as.data.frame() %>% 
  .[cell_type_markers$`Neuron-enriched`,] %>% 
  na.omit() %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           kmeans_k = 5,
           annotation_col = Q96_anno,
           annotation_colors = Q96_annoColours,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Neurons" 
  ) 
dev.off()

png("output/zebrafish cell type proportions/q96 Astrocytes.png", width = 25, height = 10, units = "cm", res = 800) 
Q96_dge %>% 
  cpm(log=TRUE) %>% 
  as.data.frame() %>% 
  .[cell_type_markers$`Astrocyte-enriched`,] %>% 
  na.omit() %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           kmeans_k = 5,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           annotation_col = Q96_anno,
           annotation_colors = Q96_annoColours,
           main = "Astrocytes" 
  )
dev.off()

png("output/zebrafish cell type proportions/q96 Microglia.png", width = 25, height = 10, units = "cm", res = 800) 
Q96_dge %>% 
  cpm(log=TRUE) %>% 
  as.data.frame() %>% 
  .[cell_type_markers$`Microglia-enriched`,] %>% 
  na.omit() %>% 
  pheatmap(color = viridis_pal(option = "magma", end = 0.9)(100), 
           border_color = NA,
           kmeans_k = 5,
           annotation_col = Q96_anno,
           annotation_colors = Q96_annoColours,
           cellheight = 15, cellwidth = 15,
           fontsize = 8, 
           main = "Microglia" 
  )
dev.off()
```










